google-tests/unit-tests/simulation_runner/native_runner/native_runner_test.cpp

```diff
@@ -504,3 +539,68 @@ TEST(NativeRunner, StatusAndCancel)
     std::cout << "Time for run: " << dur.count() << std::endl;
     std::cout << "Progress before cancel: " << prog << std::endl;
 }
+
+TEST(NativeRunner, PowerTowerTest)
+{
+    // Pulling in path variable from CMake and creating path to .stinput sample file
+    std::string sample_path = std::string(PROJECT_DIR) + std::string("/Power-tower-surround_singlefacet.stinput");
+
+    // Create Simuluation Data
+    SimulationData sd;
+
+    // Constants
+    const uint_fast64_t NRAYS = 10000;
+    const double TOL = 1e-4;
+
+    // Read Input File
+    bool success = sd.import_from_file(sample_path);
+    EXPECT_TRUE(success);
+    EXPECT_TRUE(sd.get_number_of_elements() > 0);
+    EXPECT_TRUE(sd.get_number_of_ray_sources() > 0);
+
+    std::cout << "Num Elements: " << sd.get_number_of_elements() << std::endl;
+
+    // Parameters
+    SimulationParameters &params = sd.get_simulation_parameters();
```

Comment:
Is there a reason why should not put number of threads in the simulation parameters?